### PR TITLE
Enhancement/error handling variance/12

### LINKED ISSUE #12: handle different errors
- [x] already checked out
- [x] system error
- [x] cannot be checked out
- [x] other??

### PR DESCRIPTION
**Pull request by dnoneill**
closes #12 
In order to grab information from a dictionary for the reserves/reference information it would required another get statement.  Because these items already throw error codes if they are attempted to be checked out, I just added error handling for information and did not include an extra get statement so the program can run faster.  This can be changed.

### CODE REVIEW

File: selfcheckout.py [hunk 1]
```diff
@@ -36,7 +36,8 @@ def login(userid):
     if response.status_code == 200:
         return Response(response, mimetype="application/json")
     else:
-        return response
+        return Response('Incorrect Login<br>Try Again', 500)
```
Comment (joshuago78) [inline selfcheckout.py]: This should probably be a 401 Response code. See here: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes

File: selfcheckout.py [hunk 2]
```diff
@@ -54,9 +54,15 @@ def loan(userid, barcode):
     #del params['item_barcode']
     loans_response = requests.get(url, params=params)
     already_checked_out = loans_response.json().get('item_loan', False)
+    #error handling
     if already_checked_out:
-        return Response('Already Checked Out', 409)
-    
+        return Response('This item is already checked out', 409)
+    if loans_response.status_code == 404:
+    	return Response('Error incorrect barcode', 409)
```
Comment (joshuago78) [inline selfcheckout.py]: This one should probably stay as a 404

File: selfcheckout.py [hunk 3]
```diff
@@ -54,9 +54,15 @@ def loan(userid, barcode):
...
+    if "reference" in redirect.text.lower():
+    	return Response('Cannot Checkout: Reference Materials', 409)
+    if "non-circulating" in redirect.text.lower():
+    	return Response('Cannot Checkout: Reserve Materials', 409)
```
Comment (joshuago78) [inline selfcheckout.py]: These last two should probably be 403

File: static/js/self-check.js [hunk 4]
```diff
@@ -154,11 +154,15 @@ function loan() {
...
     	}).fail(function(jqxhr, textStatus, error) {
     		console.log(jqxhr.responseText);
-            console.log(textStatus);
-            console.log(error);
-    		
+        console.log(textStatus);
+        console.log(error);
     		$("#modalheader").text("");
-    		$("#modalheader").append("item not avaiable for loan.<br/><br/>please see the reference desk for more information<br/><br/><input class='modalclose' type='button' value='close' id='barcodeerrorbutton' onclick='javascript:returnToBarcode();'/>");
+    		if (jqxhr.status == 409) {
```
Comment (joshuago78) [inline static/js/self-check.js]: By changing the response codes in the Python code, you can use them here to present different messages to the user.

**Review by joshuago78 — COMMENTED** [review]
Change error codes as specified with inline comments

File: static/js/self-check.js (hunk 4 above)
Comment (dnoneill) [reply static/js/self-check.js]: Yeah this is something I want to talk to you about I changed them to all the same for simplicity, but I can change it back.

File: selfcheckout.py (hunk 2 above)
Comment (dnoneill) [reply selfcheckout.py]: This one can't be a 404, if you put a slash in the barcode it will return a strange error code about going to an html page.

**Pull request merged**